docs/source/tutorial/trainer.rst

```diff
@@ -0,0 +1,307 @@
+
+This example will show how to use the :class:`~chainer.training.Trainer` to train a fully-connected feed-forward neural network on the MNIST dataset.
+
+.. note::
+
+    If you would like to know how to write a training loop without using this functionality, please check :doc:`tutorial/train_loop.rst` instead of this tutorial.
+
+1. Prepare the dataset
+''''''''''''''''''''''
+
+Load the MNIST dataset, which contains a training set of images and class labels as well as a corresponding test set.
+
+.. testcode::
+
+    from chainer.datasets import mnist
+
+    train, test = mnist.get_mnist()
+
+.. note::
+
+    **You can use a Python list as a dataset.** Because all types of objects whose element can be accessed via ``[]`` accessor and lengh can be obtained with ``len()`` function, can be used as a dataset given to the :class:`~chainer.dataset.Iterator`. For example,
+
+    .. code-block:: python
+
+        train = [(x1, t1), (x2, t2), ...]
+
+    a list of tuples like this can also be used equally to a :class:`~chainer.dataset.DatasetMixin` object.
+
+    But many useful abstracted :mod:`~chainer.datasets` enable to avoid storing all data on the memory at a time, so it's better to use them for large datasets. For example, :class:`~chainer.datasets.ImageDataset` takes paths to image files as its argument, and just keep the list in the dataset object. It means the actual image data will be loaded from disks using given paths when :meth:`~chainer.datasets.ImageDataset.__getitem__` is called. Until then, no images are loaded to the memory, so it can save the memory consumption.
+
+
+2. Prepare the dataset iterations
+'''''''''''''''''''''''''''''''''
+
+:class:`~chainer.dataset.Iterator` creates a mini-batch from the given dataset.
+
+.. testsetup:: *
+
+    train, test = mnist.get_mnist()
+
+.. testcode::
+
+    batchsize = 128
+
+    train_iter = iterators.SerialIterator(train, batchsize)
+    test_iter = iterators.SerialIterator(test, batchsize, False, False)
+
+3. Prepare the model
+''''''''''''''''''''
+
+Here, we are going to use the same model as defined in :doc:`tutorial/train_loop.rst`.
+
+.. testcode::
+
+    class MLP(Chain):
+
+        def __init__(self, n_mid_units=100, n_out=10):
+            super(MLP, self).__init__()
+            with self.init_scope():
+                self.l1 = L.Linear(None, n_mid_units)
+                self.l2 = L.Linear(None, n_mid_units)
+                self.l3 = L.Linear(None, n_out)
+
+        def __call__(self, x):
+            h1 = F.relu(self.l1(x))
+            h2 = F.relu(self.l2(h1))
+            return self.l3(h2)
+
+    gpu_id = 0  # Set to -1 if you use CPU
+
+    model = MLP()
+    model.to_gpu(gpu_id)  # If you use CPU, comment out this line
+
+4. Prepare the Updater
+''''''''''''''''''''''
+
+:class:`~chainer.tarining.Trainer` is a class that holds all of the necessary components needed for training. The main components are shown below.
+
+.. image:: ../../image/trainer/trainer.png
```

Comment:
This diagram can be mistaken as venn's diagram.
How about standard UML diagram with composition/aggregation relationship?
With UML, you can also represent cardinality (for example, trainer can have 0 or more extensions).